notification/app/services/NotificationConfiguration.scala

```diff
@@ -32,7 +33,10 @@ final class NotificationConfiguration @Inject()(wsClient: WSClient)(implicit exe
 
   private def notificationHub = notificationHubOR.fold(error => throw new Exception(error.message), identity)
 
-  def notificationHubClient = new NotificationHubClient(notificationHub, wsClient)
+  val notificationReportRepository = new InMemoryNotificationReportRepository
```

Comment:
If we want to have a state on that object, I would suggest that NotificationConfiguration use the @Singleton tag.
That being said I would prefer our state to be outside of the configuration (and ideally injected)